src/plugins/plugin.legend.js

```diff
@@ -411,15 +412,30 @@ var Legend = Element.extend({
 			// Horizontal
 			var isHorizontal = me.isHorizontal();
 			if (isHorizontal) {
+				// default to center as this is the original behaviour of just position
+				var alignedX = ((legendWidth - lineWidths[0]) / 2) + labelOpts.padding;
+				if (opts.alignment === 'left') {
+					alignedX = 0;
+				} else if (opts.alignment === 'right') {
+					alignedX = ((legendWidth - lineWidths[0])) + labelOpts.padding;
+				}
 				cursor = {
-					x: me.left + ((legendWidth - lineWidths[0]) / 2) + labelOpts.padding,
+					x: me.left + alignedX,
 					y: me.top + labelOpts.padding,
 					line: 0
 				};
 			} else {
+				// default to top as this is the original behaviour of just position
+				var alignedY = me.top + labelOpts.padding;
+				var legendHeight = (fontSize + labelOpts.padding) * me.legendItems.length;
+				if (opts.alignment === 'bottom') {
+					alignedY = ((me.height - legendHeight));
+				} else if (opts.alignment === 'center') {
+					alignedY = (((me.height - legendHeight) / 2)) + labelOpts.padding;
+				}
 				cursor = {
 					x: me.left + labelOpts.padding,
-					y: me.top + labelOpts.padding,
+					y: alignedY,
```

Comment:
Same as the horizontal legend. As `legendHeight` includes padding on the both sides, we need to add labelOpts.padding in any case.
```javascript
				var alignedY = me.top;
 				var maxItemsPerColumn = Math.floor((me.minSize.height - labelOpts.padding) / itemHeight);
                var legendHeight = Math.min(me.legendItems.length, maxItemsPerColumn) * itemHeight + labelOpts.padding;
				if (opts.alignment === 'bottom') {
 					alignedY = me.bottom - legendHeight;
				} else if (opts.alignment === 'center') {
 					alignedY = me.top + (me.height - legendHeight) / 2;
				}
 				cursor = {
 					x: me.left + labelOpts.padding,
 					y: alignedY + labelOpts.padding,
```